pkg/agent/core/state_test.go

```diff
@@ -234,6 +253,37 @@ func duration(s string) time.Duration {
 	return d
 }
 
+func zeroRev(t time.Time) vmv1.RevisionWithTime {
+	return vmv1.ZeroRevision.WithTime(t)
+}
```

Comment:
FYI there's some places where you use this function, and some where you use `vmv1.ZeroRevision.WithTime` explicitly. I have no preference, although `zeroRev` has less syntax :)

Reply:
That said -- I see there's a lot of places where you're passing `clock.Now()`. Maybe worthwhile to define a local anonymous function that does that automatically?

e.g.
```go
clock := ...
expectedRevision := ...

zeroRev := func() vmv1.RevisionWithTime {
    return expectedRevision.WithTime(clock.Now())
}
```

Reply:
Check again there is no expectedRevision.Revision